pkg/operator/controller/nodeobservabilityrun/controller.go

```diff
@@ -80,114 +80,105 @@ func (r *NodeObservabilityRunReconciler) Reconcile(ctx context.Context, req ctrl
 
-	if inProgress(instance) {
-		r.Log.V(1).Info("Run is in progress")
+	if runInProgress(instance) {
+		r.Log.V(1).Info("run is in progress")
 		var requeue bool
-		requeue, err = r.handleInProgress(instance)
-		if requeue {
-			msg = "Profiling query in progress"
-			instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionFalse, nodeobservabilityv1alpha2.ReasonInProgress, msg)
+		if requeue, err = r.handleInProgress(instance); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to poll agents: %w", err)
+		} else if requeue {
+			instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionFalse, nodeobservabilityv1alpha2.ReasonInProgress, "Profiling query in progress")
 			return ctrl.Result{RequeueAfter: pollingPeriod}, err
 		}
+		r.Log.V(1).Info("profiling query is done")
 		t := metav1.Now()
 		instance.Status.FinishedTimestamp = &t
-		msg = "Profiling query done"
-		instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionTrue, nodeobservabilityv1alpha2.ReasonFinished, msg)
-		return
+		instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionTrue, nodeobservabilityv1alpha2.ReasonFinished, "Profiling query done")
+		return ctrl.Result{}, nil
 	}
 
-	err = r.startRun(ctx, instance)
-	if err != nil {
-		msg = fmt.Sprintf("Failed to initiate profiling query: %s", err.Error())
-		instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionFalse, nodeobservabilityv1alpha2.ReasonFailed, msg)
-		return ctrl.Result{}, err
+	r.Log.V(1).Info("ready to initiate profiling query")
+	if err := r.startRun(ctx, instance); err != nil {
+		instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionFalse, nodeobservabilityv1alpha2.ReasonFailed, fmt.Sprintf("Failed to initiate profiling query: %s", err))
+		return ctrl.Result{}, fmt.Errorf("failed to initiate profiling query: %w", err)
 	}
 
-	msg = "Profiling query initiated"
-	instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionFalse, nodeobservabilityv1alpha2.ReasonInProgress, msg)
+	r.Log.V(1).Info("profiling query initiated")
+	instance.Status.SetCondition(nodeobservabilityv1alpha2.DebugFinished, metav1.ConditionFalse, nodeobservabilityv1alpha2.ReasonInProgress, "Profiling query initiated")
 
 	// one cycle takes cca 30s
-	return ctrl.Result{RequeueAfter: time.Second * 30}, err
+	return ctrl.Result{RequeueAfter: time.Second * 30}, nil
 }
 
-// adoptResource sets OwnerReference to point to NodeObservability from .spec.ref.name
-// returns true if already adopted, false otherwise
+// adoptResource sets OwnerReference to point to NodeObservability from .spec.nodeobservabilityref.name.
+// returns true if was adopted by this function, false otherwise.
```

Comment:
// returns true if it was adopted by this function, false otherwise.

Reply:
Got it, I missed `it`, I'm wondering whether the way it's now is a invalid English.
Will add `it` if some other comments will pop up.